File: fe/src/types/Sprint.ts
import { Issue } from "./Issue";

export enum SprintStatus {
  ACTIVE = "ACTIVE",
  PLANNED = "PLANNED",
  COMPLETED = "COMPLETED",
  CANCELED = "CANCELLED",
  BACKLOG = "BACKLOG",
}

export type Sprint = {
  projectId: number;
  id: number;
  startDate: Date;
  endDate: Date;
  status: SprintStatus;
  issues: Issue[];
};

export type OverdueSprint = {
  projectId: number;
  id: number;
  overdue: number;
};

File: be/src/main/java/com/ffb/be/model/entity/Sprint.java
package com.ffb.be.model.entity;


import com.ffb.be.model.enums.SprintStatus;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
public class Sprint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    @Enumerated(EnumType.STRING)
    private SprintStatus sprintStatus = SprintStatus.PLANNED;

    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;

    @OneToMany(mappedBy = "sprint")
    private List<Issue> issues = new ArrayList<>();

    @Override
    public String toString() {
        return "Sprint{" +
                "id=" + id +
                ", startDate=" + startDate +
                ", endDate=" + endDate +
                ", sprintStatus=" + sprintStatus +
                ", projectId=" + (project != null ? project.getId() : null) +
                ", issuesCount=" + (issues != null ? issues.size() : 0) +
                '}';
    }

}